### PR TITLE
:sparkles: add plotly-resampler as plotting engine

### PR DESCRIPTION
**Pull request by jvdd**
This PR adds `"plotly-resampler"` as option for the plotting `engine` of `StatsForecast.plot` method, see #342 

Would love to hear your feedback on this!

---

Example use of plotly-resampler (in the ElectricityLoadForecasting.ipynb notetbook)

image

Note that plotly-resampler properties can be passed in the `resampler_kwargs` argument. 
:arrow_down: illustrates how for example the number of shown samples can be changed!

```py
StatsForecast.plot(df, engine="plotly-resampler", resampler_kwargs={"default_n_shown_samples": 3000, "show_dash": {"mode": "inline"}})
```

---

PS: also fixed a minor typo in the CONTRIBUTING.md file

### CODE REVIEW

**Comment by jvdd** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jvdd** [comment]
recheck

**Comment by jvdd** [comment]
recheck

File: environment.yml
Comment (FedericoGarza) [inline environment.yml:19]: Could you please include `plotly-resampler` as a conda dependency instead of `pip`? Thank you!

**Review by FedericoGarza — CHANGES_REQUESTED** [review]
Thank you @jvdd! Nice integration! 

I have left a comment, once it is resolved, it will be ready to merge 🚀🤗

File: environment.yml
Comment (jvdd) [reply environment.yml:19]: I am fine with adding the [conda-forge feedstock](https://github.com/conda-forge/plotly-resampler-feedstock) of plotly-resampler.
However, I initially added the `pip` dependency, as 
* (@jonasvdd and me) are the maintainers of the pypi deploy - which is not the case the conda-forge deploy
* the conda-forge deploy gets updated less frequent (is still 2 versions behind the current pypi version)

So this makes me curious why this makes you prefer `conda` over `pip`? :eyes: 

P.S.: I have little to no experience with conda (for this project I created my very first conda env :upside_down_face: )

Comment (FedericoGarza) [reply environment.yml:19]: hey @jvdd! The main reason behind adding the library to conda is just as a sanity check. Since we are maintaining the [conda-forge/statsforecast](https://github.com/conda-forge/statsforecast-feedstock) repo, we must add the new dependency to the [recipe](https://github.com/conda-forge/statsforecast-feedstock/blob/main/recipe/meta.yaml) file to prevent errors or conflicts building the library. 

The alternative that comes to my mind (since you're not maintaining the conda-forge/plotly-resampler) is to try to import the library only when `engine=="plotly-resampler"` and throw an error if it is not installed.  With that change, we can include `plotly-resampler` as a core dependency in pip even if we don't include it as part of conda-forge/statsforecast. 

I'll be glad to read your thoughts on that.

Comment (jvdd) [reply environment.yml:19]: I like the alternative you suggested. I've made the necessary updates to the PR based on your suggestion.

As for the conda-forge/plotly-resampler package, I plan to become a maintainer in the future. Once I'm a maintainer, we can consider updating the dependency to conda-forge (as there will be more guarantees about the maintenance etc.).

Thanks for your feedback and suggestion!

**Comment by jvdd** [comment]
I'm not sure why the [macOS - Python 3.8](https://github.com/Nixtla/statsforecast/actions/runs/3649033414/jobs/6183348089) CI is stuck at the `Run ray` step in this pull request. It looks like the same issue occurred in #358 as well.

**Comment by jvdd** [comment]
Not really sure if this is related to the macOS test hanging, but GitHub is performing some maintenance on their macOS virtual environments 

https://github.com/actions/runner-images/issues/2247

**Review by FedericoGarza — APPROVED** [review]
Thank you @jvdd!🎉

**Comment by FedericoGarza** [comment]
@all-contributors please add @jvdd for code

**Pull request merged**
